installers/AppVeyor.hs

```diff
@@ -0,0 +1,160 @@
+
+n :: Format r (Name a -> r)
+n = makeFormat untagName
+
+ss :: Format r ([Text] -> r)
+ss = makeFormat (T.intercalate ",")
+
+----------------------------------------------------------------------------
+
+-- | Given a build status URL, query AppVeyor for job IDs.
+appVeyorJobs :: GH.URL -> IO [Job]
+appVeyorJobs buildURL = do
+  let Just req = buildAPI buildURL
+  r <- asValue =<< get (show req)
+  pure (r ^.. responseBody . buildJobs)
+
+-- | Download all artifacts for all job ids.
+appVeyorArtifacts :: [Text] -> IO [(Text, L8.ByteString)]
+appVeyorArtifacts = fmap concat . mapM artifactForJobId
+
+-- | Filter the artifact we are interested in.
+findAppVeyorArtifact :: Text -> [(Text, L8.ByteString)] -> Maybe L8.ByteString
+findAppVeyorArtifact name arts = head [bs | (name', bs) <- arts, name' == name]
+
+-- | Downloads all artifacts for a job.
+-- Returns the filename and contents.
+artifactForJobId :: Text -> IO [(Text, L8.ByteString)]
+artifactForJobId jobId = do
+  let req = jobArtifactsURI jobId
+  r <- asValue =<< get req
+  forM (r ^.. responseBody . artifactFilenames) $ \filename -> do
+    let req' = req ++ ('/':T.unpack filename)
+    printf ("Downloading "%w%" ... ") req'
+    r' <- get req'
+    printf "done\n"
+    pure (filename, r' ^. responseBody)
+
+-- | Looks into AppVeyor API build info json to get job IDs
+buildJobs :: Monoid r => Getting r Value Job
+buildJobs = key "build" . key "jobs" . _Array . traverse . _JSON
+
+-- | Looks into AppVeyor API job artifacts json to get the filenames
+artifactFilenames :: Monoid r => Getting r Value Text
+artifactFilenames = _Array . traverse . key "fileName" . _String
+
+-- | The AppVeyor API URL of an build, given its web URL.
+buildAPI :: GH.URL -> Maybe URI
+buildAPI = parseAbsoluteURI . T.unpack . T.replace "/project/" "/api/projects/" . getUrl
+
+-- | The AppVeyor API URL to list artifacts belonging to a job.
+jobArtifactsURI :: Text -> String
+jobArtifactsURI jobId = "https://ci.appveyor.com/api/buildJobs/" <> T.unpack jobId <> "/artifacts"
+
+-- | Use GitHub API to find AppVeyor build status
+appVeyorURL :: CardanoSource -> IO GH.URL
+appVeyorURL src = fmap collect <$> statusFor' src >>= \case
+  Right (Just u) -> pure u
+  Right Nothing  -> throwM $ StatusMissingError src
+  Left err       -> throwM $ GitHubStatusError src err
+  where
+    collect = head . mapMaybe statusTargetUrl . filter isAppVeyor .
+              toList . combinedStatusStatuses
+    isAppVeyor st = statusContext st == Just "continuous-integration/appveyor/branch"
+
+statusFor' :: CardanoSource -> IO (Either GH.Error GH.CombinedStatus)
+statusFor' CardanoSource{..} = statusFor noAuth srcOwner srcRepo srcRev
+  where noAuth = BasicAuth "" ""
+
+data JobStatus = JobSuccess | JobFailed | JobRunning
+               | JobUnknown Text deriving (Show, Eq, Generic)
+
+data Job = Job { jobId :: Text
+               , jobStatus :: JobStatus
+               } deriving (Show, Eq, Generic)
```

Comment:
Do you think it might be worth moving type definitions into `Types.hs`?

We already have a `BuildJob` type there (yeah, kind of "fallout" from `DEVOPS-690` being merged first..): https://github.com/input-output-hk/daedalus/blob/develop/installers/Types.hs#L69

Reply:
..and this actually reminds me that we actually have a bunch of very similar pre-existing types in `iohk-ops`..

Maybe we should have a discussion (outside of this PR!) about having them merged together..

Reply:
I already brought this up at the standup.